File: C++/Estrutura Repetitiva/Para/Experiencias.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <climits>
#include <cmath>

using namespace std;

int main()
{

    int N, i, qtde, coelho, rato, sapo, total;
    char tipo;
    double percC, percR, percS;

    cout << "Quantos casos de teste serao digitados? ";
    cin >> N;

    coelho = 0;
    rato = 0;
    sapo = 0;
    for (i = 1; i <= N; i++)
    {
        cout << "Quantidade de cobaias? ";
        cin >> qtde;
        cout << "Tipo da cobaia: ";
        cin >> tipo;
        if (tipo == 'C')
        {
            coelho = coelho + qtde;
        }
        else if (tipo == 'R')
        {
            rato = rato + qtde;
        }
        else
        {
            sapo = sapo + qtde;
        }
    }

    cout << endl;
    total = coelho + rato + sapo;
    percC = (double)coelho / total * 100;
    percS = (double)sapo / total * 100;
    percR = (double)rato / total * 100;

    cout << "RELATORIO FINAL: " << endl;
    cout << "Total: " << total << " cobaias" << endl;
    cout << "Total de coelhoe: " << coelho << endl;
    cout << "Total de ratos: " << rato << endl;
    cout << "Total de sapos: " << sapo << endl;
    cout << fixed << setprecision(2);
    cout << "Percentual de coelhos: " << percC << endl;
    cout << "Percentual de ratos: " << percR << endl;
    cout << "Percentual de sapos: " << percS << endl;
    

    return 0;
}

File: Python/Estrutura Repetitiva/Para/Experiencias.py
N: int; i: int; x: int; coelho: int; rato: int; sapo: int; total: int
tipo: str
percRato: float; percSapo: float; percCoelho: float



N = int(input("Quantos casos de teste serao digitados? "))

coelho = 0
sapo = 0
rato = 0

for i in range(1, N + 1):
    x = int(input("Quantidade de cobaias: "))
    tipo = str(input("Tipo de cobaia: "))
    if tipo == 'C':
        coelho = coelho + x
    elif tipo == 'R':
        rato = rato + x
    elif tipo == 'S':
        sapo = sapo + x

total = coelho + rato + sapo
percCoelho = coelho / total * 100
percRato = rato / total * 100
percSapo = sapo / total * 100

print(f"RELATORIO FINAL:")
print(f"Total: {total} cobaias")
print(f"Total de coelhos: {coelho}")
print(f"Total de ratos: {rato}")
print(f"Total de sapos: {sapo}")
print(f"Percentual de coelhos: {percCoelho:.2f}")
print(f"Percentual de ratos: {percRato:.2f}")
print(f"Percentual de sapos: {percSapo:.2f}")
